core/remotes/docker/config/hosts.go

```diff
@@ -228,6 +230,14 @@ func ConfigureHosts(ctx context.Context, options HostOptions) docker.RegistryHos
 					tlsConfig.Certificates = append(tlsConfig.Certificates, cert)
 				}
 
+				if host.dialTimeout != nil {
+					tr.DialContext = (&net.Dialer{
+						Timeout:       *host.dialTimeout,
+						KeepAlive:     30 * time.Second,
+						FallbackDelay: 300 * time.Millisecond,
+					}).DialContext
+				}
```

Comment:
This will only take effect if line 184 is true

Reply:
Also, we should set the current default value(`30 sec`) if `host.dialTimeout` has not been set.